salt/serializers/keyvalue.py

```diff
@@ -0,0 +1,119 @@
+"""
+    salt.serializers.keyvalue
+    ~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Implements keyvalue serializer which can be used for serializing or
+    deserializing any file which defines keys and values separated by a common
+    set of characters, such environment files, which are in "KEY=value" format.
+
+    Options:
+
+    :param line_ending:
+        String representation of LF or CRLF to be used for serialization to a
+        file. Defaults to ``\\r\\n`` on Windows and ``\\n`` on other operating
+        systems.
+
+    :param quoting:
+        Boolean flag to determine if values should be quoted (``True``) during
+        serialization or dequoted (``False``) during deserialization.
```

Comment:
This doesn't seem right. Surely when `True` it should both quote when serializing and dequote when deserializing?

When `False` it should write and read raw values.

Reply:
It also doesn't say what the default is.

Reply:
According to the tests it also uses single quotes. There should probably be an option for double quotes.